### ISSUE TITLE
Git Bash - Process Substitution does not work - /dev/fd/62: Read-only file system

### ISSUE THREAD

**Issue by dnut** [git-for-windows/git#3105]
- [x] I was not able to find an [open](https://github.com/git-for-windows/git/issues?q=is%3Aopen) or [closed](https://github.com/git-for-windows/git/issues?q=is%3Aclosed) issue matching what I'm seeing

### Setup

 - Which version of Git for Windows are you using? Is it 32-bit or 64-bit?

```
$ git --version --build-options

git version 2.23.0.windows.1
cpu: x86_64
built from commit: 4db2e5cc9e1522131a039cbad3970f147a39f0ce
sizeof-long: 4
sizeof-size_t: 8
```

 - Which version of Windows are you running? Vista, 7, 8, 10? Is it 32-bit or 64-bit?

```
$ cmd.exe /c ver

Microsoft Windows [Version 10.0.19042.746]
```

 - What options did you set as part of the installation? Or did you choose the
   defaults?

```
$ cat /etc/install-options.txt

Editor Option: VIM
Custom Editor Path:
Path Option: Cmd
SSH Option: OpenSSH
Tortoise Option: false
CURL Option: OpenSSL
CRLF Option: CRLFAlways
Bash Terminal Option: MinTTY
Performance Tweaks FSCache: Enabled
Use Credential Manager: Enabled
Enable Symlinks: Disabled
Enable Builtin Interactive Add: Disabled
```

### Details

 - Which terminal/shell are you running Git from?

Git Bash in Windows Terminal.

 - What commands did you run to trigger this issue?

```bash
echo hi > >(cat)
```
 - What did you expect to occur after running these commands?

The shell should print "hi"

 - What actually happened instead?

```
bash: /dev/fd/62: Read-only file system
```

This is a trivial example to prove a point. I run into the same problem with any process substitution.

**Comment by rimrul** [comment]
This should be fixed in version 2.24.0 or newer. See #2291

**Comment by dscho** [comment]
> git version 2.23.0.windows.1

This is quite old, and open to a plethora of vulnerabilities (see https://github.com/git-for-windows/git/wiki/FAQ#should-i-upgrade-to-a-newer-git-for-windows-version for details). Please do upgrade.
